pkg/test/ginkgo/cmd_runsuite.go

```diff
@@ -285,9 +292,13 @@ func (opt *Options) Run(suite *TestSuite) error {
 	// TODO: will move to the monitor
 	pc.SetEvents([]string{upgradeEvent})
 
-	// repeat the normal suite until context cancel when in the forever loop
+	// run our k8s tests
+	q.Execute(testCtx, kube, parallelism, status.Run)
+	tests = append(tests, kube...)
+
+	// repeat anything not kube until context cancel when in the forever loop
 	for i := 0; (i < 1 || count == -1) && testCtx.Err() == nil; i++ {
-		copied := copyTests(normal)
+		copied := copyTests(others)
```

Comment:
I don't understand this loop (we repeatedly run tests until cancel?  that sounds strange.  What's the expected cancel event?), which i realize you didn't introduce, but it makes me wonder if the kube tests should also be inside it or not.  (without knowing what this loop is intended to achieve, it's hard for me to assess)

Reply:
Haha, I looked at this loop for a while. It's based on the `--count` arg to run the tests n times. It accomplishes that by duplicating the tests n times other than early/late.

Reply:
I think the comment is wrong, too... not sure where the forever loop is.

Reply:
that logic is...wow....  can you fix up the comment please?

i guess it runs forever(until we hit an error or presumably a timeout/interrupt) if count==-1, so that's probably what the comment means by "when in the forever loop", but the forever loop is not the default case(i assume).

Reply:
anyway i think it's still an open question whether the kube tests should also be subject to being run multiple times, @deads2k probably has the most informed opinion on that.

Reply:
I updated the comment, and put the kube tests in the loop. Let me know if I should take it out -- I wonder how often people use --count=-1 anyway